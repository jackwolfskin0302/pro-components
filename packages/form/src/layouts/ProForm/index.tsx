import React from 'react';
import { Form } from 'antd';
import { FormProps } from 'antd/lib/form/Form';
import Group from '../../components/Group';
import BaseForm, { CommonFormProps } from '../../BaseForm';

export interface ProFormProps extends FormProps, CommonFormProps {
  // ProForm 基础表单，暂无特殊属性
}

const ProForm: React.FC<ProFormProps> & {
  Group: typeof Group;
  useForm: typeof Form.useForm;
} = (props) => {
  return (
    <BaseForm
      layout="vertical"
      submitter={{
        // 反转按钮，在正常模式下，按钮应该是主按钮在前
        render: (_, dom) => dom.reverse(),
      }}
      contentRender={(items, submitter) => {
        return (
          <>
            {items}
            {submitter}
          </>
        );
      }}
      {...props}
    />
  );
};

ProForm.Group = Group;
ProForm.useForm = Form.useForm;

export default ProForm;
