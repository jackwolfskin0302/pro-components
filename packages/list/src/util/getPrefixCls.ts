const getPrefixCls = (suffixCls: string, customizePrefixCls?: string) => {
  const prefixCls = 'ant-pro';

  if (customizePrefixCls) return customizePrefixCls;

  return suffixCls ? `${prefixCls}-${suffixCls}` : prefixCls;
};

export default getPrefixCls;
