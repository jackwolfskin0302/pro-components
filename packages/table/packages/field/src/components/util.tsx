/**
 * 检查值是否存在
 * 为了 避开 0 和 false
 * @param value
 */
export const checkUndefinedOrNull = (value: any) =>
  value !== undefined && value !== null;

/**
 * 删除对象中所有的空值
 * @param obj
 */
export const removeObjectNull = (obj: { [key: string]: any }) => {
  const newObj = {};
  Object.keys(obj).forEach(key => {
    if (obj[key]) {
      newObj[key] = obj[key];
    }
  });
  return newObj;
};
