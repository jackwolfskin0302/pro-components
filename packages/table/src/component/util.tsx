/* eslint-disable @typescript-eslint/naming-convention */
import React, { ReactText } from 'react';
import { DataIndex } from 'rc-table/lib/interface';

/**
 * 检查值是否存在
 * 为了 避开 0 和 false
 * @param value
 */
export const checkUndefinedOrNull = (value: any) => value !== undefined && value !== null;

/**
 *  根据 key 和 dataIndex 生成唯一 id
 * @param key 用户设置的 key
 * @param dataIndex 在对象中的数据
 * @param index 序列号，理论上唯一
 */
export const genColumnKey = (
  key?: React.ReactText | undefined,
  dataIndex?: DataIndex,
  index?: number,
) => {
  if (key) {
    return key;
  }
  if (!key && dataIndex) {
    if (Array.isArray(dataIndex)) {
      return dataIndex.join('-');
    }
    return dataIndex;
  }
  return `${index}`;
};

/**
 * 删除对象中所有的空值
 * @param obj
 */
export const removeObjectNull = (obj: { [key: string]: any }) => {
  const newObj = {};
  Object.keys(obj).forEach((key) => {
    if (obj[key]) {
      newObj[key] = obj[key];
    }
  });
  return newObj;
};

/**
 * 减少 width，支持 string 和 number
 */
export const reduceWidth = (width?: string | number): string | number | undefined => {
  if (width === undefined) {
    return width;
  }
  if (typeof width === 'string') {
    if (!width.includes('calc')) {
      return `calc(100% - ${width})`;
    }
    return width;
  }
  if (typeof width === 'number') {
    return (width as number) - 32;
  }
  return width;
};
