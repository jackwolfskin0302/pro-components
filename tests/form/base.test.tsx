import React from 'react';
import { Button } from 'antd';
import ProForm, { ProFormText } from '@ant-design/pro-form';
import { mount } from 'enzyme';
import { waitTime } from '../util';

describe('ProForm', () => {
  it('submit props actionsRender=false', async () => {
    const wrapper = mount(<ProForm submitter={false} />);
    await waitTime();
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('submit props actionsRender=()=>false', async () => {
    const wrapper = mount(
      <ProForm
        submitter={{
          render: () => false,
        }}
      />,
    );
    await waitTime();
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('submit props actionsRender=()=>[]', async () => {
    const wrapper = mount(
      <ProForm
        submitter={{
          render: () => [],
        }}
      />,
    );
    await waitTime();
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('submit props render=()=>[]', async () => {
    const wrapper = mount(
      <ProForm
        submitter={{
          render: () => [
            <Button key="submit" type="primary">
              提交并发布
            </Button>,
          ],
        }}
      />,
    );
    await waitTime();
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('submitter.render simulate onFinish', async () => {
    const onFinish = jest.fn();
    const wrapper = mount(
      <ProForm
        onFinish={onFinish}
        submitter={{
          render: ({ form }) => [
            <Button
              id="submit"
              key="submit"
              type="primary"
              onClick={() => {
                form.submit();
              }}
            >
              提交并发布
            </Button>,
          ],
        }}
      >
        <ProFormText label="name" name="name" />
      </ProForm>,
    );
    await waitTime();

    wrapper.find('button#submit').simulate('click');
    await waitTime(100);

    expect(onFinish).toBeCalled();
  });
});
