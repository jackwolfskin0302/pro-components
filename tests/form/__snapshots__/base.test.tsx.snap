// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProForm submit props actionsRender=()=>[] 1`] = `
<form
  class="ant-form ant-form-vertical"
/>
`;

exports[`ProForm submit props actionsRender=()=>false 1`] = `
<form
  class="ant-form ant-form-vertical"
/>
`;

exports[`ProForm submit props actionsRender=false 1`] = `
<form
  class="ant-form ant-form-vertical"
/>
`;

exports[`ProForm submit props render=()=>[] 1`] = `
<form
  class="ant-form ant-form-vertical"
>
  <div
    class="ant-space ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
    >
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span>
          提交并发布
        </span>
      </button>
    </div>
  </div>
</form>
`;
